sys/kern/signal.c

```diff
@@ -95,7 +98,7 @@ int do_sigaction(signo_t sig, const sigaction_t *act, sigaction_t *oldact) {
       memcpy(&p->p_sigactions[sig], act, sizeof(sigaction_t));
     /* If ignoring a pending signal, discard it. */
     if (sig_ignored(p->p_sigactions, sig))
-      __sigdelset(&td->td_sigpend, sig);
+      __sigdelset(&td->td_sigpend.sp_set, sig);
```

Comment:
Shouldn't we also remove the queued `ksiginfo` here? If we don't and then send a `KSI_RAW` signal, the target process will receive a signal that should have been discarded.

Reply:
Fixed.